content/usgs-packages/sbtools_Assessment.Rmd

```diff
@@ -39,11 +39,72 @@ knit_hooks$set(addToggle = function(before, options, envir) {
 set.seed(1)
 ```
 
+```{r sbtools-auth, echo=FALSE}
+# run vizlab::storeSBcreds() once before this can work
+home <- path.expand('~')
+sbCreds <- file.path(home, ".vizlab/sbCreds")
+credList <- readRDS(sbCreds)
+un <- rawToChar(credList$username)
+pw <- rawToChar(credList$password)
+sbtools::authenticate_sb(un, pw)
+```
+
 Before starting the exercises, you should make sure that the `sbtools` package is installed and loaded. If you haven't recently updated, you could reinstall the package by running `install.packages('sbtools')` or go to the "Update" button in the "Packages" tab in RStudio.
 
 ```{r load-sbtools}
 # load the sbtools package
 library(sbtools)
 ```
 
+## Exercise 1
+
+Using querying functions, find out how many ScienceBase items exist related to lake temperature during 1992. Hint: use `query_sb` to search using more than one criteria.
+
+```{r addToggle="ON"}
+laketemp <- query_sb(query_list = list(q = 'lake temperature', 
+                           dateRange = '{"start":"1992-01-01","end":"1992-12-31"}'),
+                   limit = 100)
+length(laketemp)
+```
```

Comment:
I mildly prefer an answer with "lake temperature" quoted, because the current answer returns a couple of questionable results, e.g., "Thermal waters along the Konocti Bay fault zone, Lake County, California: a re-evaluation". I know everybody's pretty familiar with the weaknesses of text searching now that we've had a couple decades with the internet, but still:
If you go for the whole term "lake temperature" rather than simply requiring both words to be present somewhere, you get 0 items in 1992. So another couple of possible answers to an amended question (2010-2012 instead of 1992) give you
```r
> length(query_sb(query_list = list(q='', lq = '"lake temperatures"', dateRange = '{"start":"2010-01-01","end":"2012-12-31"}'), limit = 100))
[1] 9
> length(query_sb(query_list = list(q='', lq = '"lake temperature"', dateRange = '{"start":"2010-01-01","end":"2012-12-31"}'), limit = 100))
[1] 28
```
and i think we can reasonably expect students to come up with several acceptable answers for this question; that should be fine.